src/settings/MappingProfiles/MappingProfilesForm.js

```diff
@@ -73,127 +85,98 @@ export const MappingProfilesFormComponent = ({
 
   const folioRecordTypesDataOptions = useMemo(getFolioRecordTypesDataOptions, []);
   const incomingRecordTypesDataOptions = useMemo(getIncomingRecordTypesDataOptions, []);
-  const { layer } = queryString.parse(search);
-  const isEditMode = layer === LAYER_TYPES.EDIT;
-  const isSubmitDisabled = pristine || submitting;
-  const paneTitle = isEditMode
-    ? (
-      <FormattedMessage id="ui-data-import.edit">
-        {txt => `${txt} ${profile.name}`}
-      </FormattedMessage>
-    )
-    : <FormattedMessage id="ui-data-import.settings.mappingProfiles.new" />;
-  const headLine = isEditMode
-    ? profile.name
-    : <FormattedMessage id="ui-data-import.settings.mappingProfiles.new" />;
-  const associations = [
-    ...[],
-    ...get(initialValues, ['profile', 'parentProfiles'], []),
-    ...get(initialValues, ['profile', 'childProfiles'], []),
-  ];
 
+  const paneTitle = isEditMode ? (
+    <FormattedMessage id="ui-data-import.edit">
+      {txt => `${txt} ${profile.name}`}
+    </FormattedMessage>
+  ) : <FormattedMessage id="ui-data-import.settings.mappingProfiles.new" />;
+  const headLine = isEditMode ? profile.name : <FormattedMessage id="ui-data-import.settings.mappingProfiles.new" />;
+
+  const prevExistingRecordType = usePrevious(get(profile, 'existingRecordType', null));
+  const [existingRecordType, setExistingRecordType] = useState(get(profile, 'existingRecordType', null));
+  const [initials, setInitials] = useState({
+    ...profile,
+    mappingDetails: !mappingDetails || isEmpty(mappingDetails) ? getInitialDetails(existingRecordType, true) : mappingDetails,
```

Comment:
I think it can be simplified a bit, please give it a try.
```suggestion
    mappingDetails: isEmpty(mappingDetails) ? getInitialDetails(existingRecordType, true) : mappingDetails,
```

Reply:
Done